### PR TITLE
feat: Move packages to protocol-kit

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves ?

Resolves https://github.com/safe-global/safe-core-sdk/issues/350

## How this PR solve it

Move packages inside protocol-kit
- safe-core-sdk-utils
- safe-web3-lib
- safe-ethers-lib

### CODE REVIEW

File: packages/protocol-kit/scripts/generateTypechainFiles.ts [hunk 1]
```diff
@@ -83,30 +83,33 @@ function moveTypechainFiles(inDir: string, outDir: string): void {
   })
 }
 
-const ethersV5 = 'ethers-v5'
+function generateTypes(version: string) {
```
Comment (germartinez) [inline packages/protocol-kit/scripts/generateTypechainFiles.ts:86]: This `version` variable name can be confused with the version of a Safe, but it refers to the use of `ethers` and `web3` and their used versions.
I suggest using `typechainTarget`, which is the value expected here `typechain --target `

File: README.md [hunk 2]
```diff
@@ -8,23 +8,20 @@ Software developer tools that facilitate the interaction with the Safe [contract
...
-[api-kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/api-kit) | [![NPM Version](https://badge.fury.io/js/%40safe-global%2Fapi-kit.svg)](https://badge.fury.io/js/%40safe-global%2Fapi-kit) | [Safe Transaction Service API](https://github.com/safe-global/safe-transaction-service) client library |
-[safe-ethers-adapters](https://github.com/safe-global/safe-core-sdk/tree/main/packages/safe-ethers-adapters) | [![NPM Version](https://badge.fury.io/js/%40safe-global%2Fsafe-ethers-adapters.svg)](https://badge.fury.io/js/%40safe-global%2Fsafe-ethers-adapters) | [Ethers](https://docs.ethers.io/v5/single-page/) adapter that facilitates the interaction with the [Safe Services](https://github.com/safe-global/safe-transaction-service) |
+| Package                                                                                                      |                                                                       Release                                                                        | Description                                                                                                                                                                 |
```
Comment (germartinez) [inline README.md:17]: In general, I would remove all the extra spaces and minus symbols added to this file

**Comment by germartinez** [comment]
Good job! :)

File: guides/integrating-the-safe-core-sdk.md [hunk 3]
```diff
@@ -36,8 +30,8 @@ First of all, we need to create an `EthAdapter`, which contains all the required
 
 Depending on the library used by the Dapp, there are two options:
 
-- [Create an `EthersAdapter` instance](https://github.com/safe-global/safe-core-sdk/tree/main/packages/safe-ethers-lib#initialization)
-- [Create a `Web3Adapter` instance](https://github.com/safe-global/safe-core-sdk/tree/main/packages/safe-web3-lib#initialization)
+- [Create an `EthersAdapter` instance](https://github.com/safe-global/safe-core-sdk/blob/aa61f1e6e841594e14edb1acfee54bbf1408100b/packages/protocol-kit/src/adapters/ethers)
+- [Create a `Web3Adapter` instance](https://github.com/safe-global/safe-core-sdk/blob/aa61f1e6e841594e14edb1acfee54bbf1408100b/packages/protocol-kit/src/adapters/web3)
```
[the remark this answers is not in the document]
Comment (yagopv) [reply guides/integrating-the-safe-core-sdk.md:34]: yes, i can add the url based on `main` for all. I left it like this for not having broken links

[the remark this answers is not in the document]
Comment (yagopv) [reply guides/integrating-the-safe-core-sdk.md:34]: WDYT about setting the final url?

File: packages/protocol-kit/package.json [hunk 4]
```diff
@@ -73,16 +75,22 @@
     "ts-generator": "^0.1.1",
     "typescript": "^4.9.4",
     "web3": "^1.8.1",
```
[the remark this answers is not in the document]
Comment (yagopv) [reply packages/protocol-kit/package.json]: done

File: guides/integrating-the-safe-core-sdk.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (germartinez) [reply guides/integrating-the-safe-core-sdk.md:34]: I would say it's okay for now

[the remark this answers is not in the document]
Comment (yagopv) [reply guides/integrating-the-safe-core-sdk.md:34]: nah, you are right, we are going to forget :D. Updated the links

File: packages/api-kit/README.md [hunk 5]
```diff
@@ -40,8 +41,8 @@ First of all, we need to create an `EthAdapter`, which contains all the required
 
 Depending on the library used by the Dapp, there are two options:
 
-- [Create an `EthersAdapter` instance](https://github.com/safe-global/safe-core-sdk/tree/main/packages/safe-ethers-lib#initialization)
-- [Create a `Web3Adapter` instance](https://github.com/safe-global/safe-core-sdk/tree/main/packages/safe-web3-lib#initialization)
+- [Create an `EthersAdapter` instance](https://github.com/safe-global/safe-core-sdk/blob/aa61f1e6e841594e14edb1acfee54bbf1408100b/packages/protocol-kit/src/adapters/ethers)
+- [Create a `Web3Adapter` instance](https://github.com/safe-global/safe-core-sdk/blob/aa61f1e6e841594e14edb1acfee54bbf1408100b/packages/protocol-kit/src/adapters/web3)
```
[the remark this answers is not in the document]
Comment (yagopv) [reply packages/api-kit/README.md:45]: Updated

File: packages/protocol-kit/src/utils/index.ts [hunk 6]
```diff
@@ -6,6 +6,10 @@ import {
 } from '@safe-global/safe-core-sdk-types'
 import { SENTINEL_ADDRESS, ZERO_ADDRESS } from './constants'
 
+export * from './eip-3770'
+export * from './eip-712'
+export * from './safeVersions'
+
```
[the remark this answers is not in the document]
Comment (yagopv) [reply packages/protocol-kit/src/utils/index.ts]: done

File: README.md (hunk 2 above)
Comment (yagopv) [reply README.md]: done

File: packages/protocol-kit/src/adapters/web3/contracts/GnosisSafe/GnosisSafeContractWeb3.ts [hunk 7]
```diff
@@ -6,10 +6,10 @@ import {
...
   SafeSetupConfig
 } from '@safe-global/safe-core-sdk-types'
-import { Gnosis_safe as GnosisSafe_V1_0_0 } from '../../../typechain/src/web3-v1/v1.0.0/Gnosis_safe'
-import { Gnosis_safe as GnosisSafe_V1_1_1 } from '../../../typechain/src/web3-v1/v1.1.1/Gnosis_safe'
-import { Gnosis_safe as GnosisSafe_V1_2_0 } from '../../../typechain/src/web3-v1/v1.2.0/Gnosis_safe'
-import { Gnosis_safe as GnosisSafe_V1_3_0 } from '../../../typechain/src/web3-v1/v1.3.0/Gnosis_safe'
+import { Gnosis_safe as GnosisSafe_V1_0_0 } from '../../../../../typechain/src/web3-v1/v1.0.0/Gnosis_safe'
+import { Gnosis_safe as GnosisSafe_V1_1_1 } from '../../../../../typechain/src/web3-v1/v1.1.1/Gnosis_safe'
+import { Gnosis_safe as GnosisSafe_V1_2_0 } from '../../../../../typechain/src/web3-v1/v1.2.0/Gnosis_safe'
+import { Gnosis_safe as GnosisSafe_V1_3_0 } from '../../../../../typechain/src/web3-v1/v1.3.0/Gnosis_safe'
```
Comment (dasanra) [inline packages/protocol-kit/src/adapters/web3/contracts/GnosisSafe/GnosisSafeContractWeb3.ts:12]: We need to setup root level routes for this packages. Too much path navigations in these files.

I would like it as a separate PR

Comment (yagopv) [reply packages/protocol-kit/src/adapters/web3/contracts/GnosisSafe/GnosisSafeContractWeb3.ts:12]: yes, all the repo should have root level routing

File: packages/protocol-kit/package.json [hunk 8]
```diff
@@ -72,17 +74,22 @@
...
     "@safe-global/safe-core-sdk-types": "^1.9.0",
-    "@safe-global/safe-core-sdk-utils": "^1.7.2",
     "@safe-global/safe-deployments": "^1.20.2",
     "ethereumjs-util": "^7.1.5",
     "semver": "^7.3.8",
-    "web3-utils": "^1.8.1"
+    "web3-utils": "^1.8.1",
+    "web3": "^1.8.1",
+    "web3-core": "^1.8.1",
+    "@ethersproject/bignumber": "^5.7.0"
```
Comment (dasanra) [inline packages/protocol-kit/package.json:92]: Incorrect ordering, should be at top of the list

File: packages/protocol-kit/package.json [hunk 9]
```diff
@@ -72,17 +74,22 @@
...
-    "yargs": "^17.6.2"
+    "yargs": "^17.6.2",
+    "@types/web3": "1.0.19",
+    "@typechain/ethers-v5": "^10.2.0",
+    "@typechain/web3-v1": "^6.0.2",
+    "typechain": "^8.1.1"
   },
   "dependencies": {
     "@ethersproject/solidity": "^5.7.0",
     "@safe-global/safe-core-sdk-types": "^1.9.0",
```
[the remark this answers is not in the document]
Comment (yagopv) [reply packages/protocol-kit/package.json]: done

**Review by dasanra — APPROVED** [review]

**Pull request merged**